File: coinflip.py
import random

# Function to read bias from file
def read_bias():
    bias = 50
    try:
        with open("bias.txt", "r") as file:
            bias = int(file.read())
    except FileNotFoundError:
        pass
    return bias

# Function to simulate a biased coin flip
def coin_flip(bias):
    random_val = random.randint(0, 100)
    if random_val > bias:
        print("tails")
    else:
        print("heads")
    return random_val > bias

def main():
    bias = read_bias()

    while True:
        first_flip = coin_flip(bias)
        second_flip = coin_flip(bias)
        if first_flip and not second_flip:
            print("\nPerson 1 wins!")
            break
        elif not first_flip and second_flip:
            print("\nPerson 2 wins!")
            break
        else:
            print("Toss discarded, flipping again.\n")

main()

File: Coinflip.cpp
#include <iostream>
#include <fstream>
#include <stdlib.h>
using namespace std;


// Function to read bias from file
double read_bias() {
    int bias = 50;
    ifstream myfile ("bias.txt");
    myfile >> bias;
    return bias;
}

// Function to simulate a biased coin flip
bool coin_flip(double bias) {
    int randoml = rand()%100;
    if (randoml > bias){
        cout << "tails" << endl;
    } else {
        cout << "heads" << endl;
    }

    return randoml > bias;
}

int main() {
    double bias = read_bias();
    
    bool first_flip, second_flip;
    do {
        first_flip = coin_flip(bias);
        second_flip = coin_flip(bias);
        if (first_flip && !second_flip) {
            cout << "Person 1 wins!" << endl;
        } else if (!first_flip && second_flip) {
            cout << "Person 2 wins!" << endl;
        } else {
            cout << "Toss discarded, flipping again.\n" << endl;
        }
    } while (first_flip == second_flip); // Redo if both flips are the same
    
    return 0;
}

